src/uiSelectController.js

```diff
@@ -454,9 +454,16 @@ uis.controller('uiSelectCtrl',
   };
 
   ctrl.clear = function($event) {
+    var locals = {};
+    locals[ctrl.parserResult.itemName] = ctrl.selected;
+
     ctrl.select(null);
     $event.stopPropagation();
     $timeout(function() {
+      ctrl.onRemoveCallback($scope, {
```

Comment:
I mean then the `clear` function. As the `uiSelectMultiple` has his own `remove` function. I think it would be good to keep  this kind of functionality on the correct place. What do you think?